pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -654,6 +654,7 @@ func (aws *AWSCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
 		}
 		return []api.NodeAddress{
 			{Type: api.NodeInternalIP, Address: internalIP},
+			{Type: api.NodeLegacyHostIP, Address: internalIP},
```

Comment:
Actually, on second thoughts I think I am wrong here.  The special case here should match the normal case below, so your change is good.